src/command.rs

```diff
@@ -239,73 +316,63 @@ pub mod file_error {
     impl std::error::Error for Error {}
 }
 
-impl Init {
-    pub fn new() -> Result<Init, Box<dyn std::error::Error>> {
-        let user = user::User::new()?;
-
-        Ok(Init { user: user })
-    }
-
-    pub fn user(&self) -> &user::User {
-        &self.user
-    }
-}
-
-impl<D> Launch<D>
-where
-    D: image_downloader::Downloader,
-{
-    pub fn new(
-        target_container: String,
-        rootfs_option: RootFSOption<D>,
-        name: String,
-        cmd: Option<String>,
-    ) -> Launch<D> {
-        Launch {
-            target_container: target_container,
-            rootfs_option: rootfs_option,
+impl File {
+    pub fn new(name: String, from: PathBuf, to: PathBuf) -> File {
+        File {
             name: name,
-            cmd: cmd,
+            from: from,
+            to: to,
         }
     }
 }
 
-impl List {
-    pub fn new() -> Result<List, Box<dyn std::error::Error>> {
-        let user_info = user::User::new()?;
+/// デバックコンテナ内で実行するコマンドを表す構造体
+/// コンストラクタの引数として何も指定されていない場合は`/bin/bash`がデフォルトで用いられる
+/// ```ignore
+/// let cmd_vec = vec![
+///     String::from("echo"),
+///     String::from("hoge"),
+/// ];
+///
+/// let cmd = Cmd::new(cmd_vec)
+/// ```
+#[derive(Debug)]
+pub struct Cmd {
+    /// mainはexecシステムコールの第1引数を表す。
+    /// `echo hogehoge`の場合は`echo`が入る
+    main: String,
+    /// execシステムコールの第２引数以降が入る
+    /// `echo hoge`の場合は`echo`, `hoge`が入る
+    detail: Vec<String>,
+}
 
-        Ok(List { user: user_info })
-    }
+impl Cmd {
+    pub fn new(mut detail: Box<dyn Iterator<Item = String>>) -> Cmd {
+        let main = match detail.next() {
+            Some(cmd) => cmd.to_string(),
+            None => "/bin/bash".to_string(),
+        };
```

Comment:
`parser::LaunchArgs`のcmdメンバは`Vec<String>`なので、
それをそのままnew()の引数に渡し、new()の中でiterに変換すればBoxなど使わなくて済むのでは？

Reply:
また、`String.to_string()`になっているのを解消する

```suggestion
impl Cmd {
    pub fn new(mut detail: Box<dyn Iterator<Item = String>>) -> Cmd {
        let main = match detail.next() {
            Some(cmd) => cmd,
            None => "/bin/bash".to_string(),
        };
```